v-next/hardhat/src/internal/builtin-plugins/network-manager/hook-handlers/hre.ts

```diff
@@ -47,7 +46,7 @@ async function createNetworkManager(
     hre.config.networks,
     context.hooks,
     context.artifacts,
-    userConfigNetworks,
```

Comment:
To fully run the config resolution hooks that will extend the config for NetworkConfig, we need to run all the plugins across all of the config.
Hence we now pass in the the original `userConfig` and not just its networks.